File: students/students/src/MON.SPA/src/models/studentSanctionModel.js
import Constants from '@/common/constants.js';
import { DocumentModel } from '@/models/documentModel.js';

export class StudentSanctionModel {
  constructor(obj = {}, moment = {}) {
    this.id = obj.id || null;
    this.startDate = (obj.startDate && moment) ? moment(obj.startDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.endDate = (obj.endDate && moment) ? moment(obj.endDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.personId = obj.personId || null;
    this.description = obj.description || null;
    this.sanctionTypeId = obj.sanctionTypeId || null;
    this.institutionId = obj.institutionId || null;
    this.cancelOrderDate = (obj.cancelOrderDate && moment) ? moment(obj.cancelOrderDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.cancelOrderNumber = obj.cancelOrderNumber || null;
    this.cancelReason = obj.cancelReason || null;
    this.orderDate = (obj.orderDate && moment) ? moment(obj.orderDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.orderNumber = obj.orderNumber || null;
    this.ruoOrderDate = (obj.ruoOrderDate && moment) ? moment(obj.ruoOrderDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.ruoOrderNumber = obj.ruoOrderNumber || null;
    this.schoolYear = obj.schoolYear;
    this.schoolYearName = obj.schoolYearName;
    this.institutionName = obj.institutionName;
    this.documents = obj.documents ? obj.documents.map(el => new DocumentModel(el)) : [];
  }
}

File: students/students/src/MON.Models/StudentModels/StudentSanctionModel.cs
namespace MON.Models.StudentModels
{
    using System;
    using System.Collections.Generic;

    public class StudentSanctionModel
    {
        public int? Id { get; set; }
        public string OrderNumber { get; set; }
        public DateTime OrderDate { get; set; }
        public string RuoOrderNumber { get; set; }
        public DateTime? RuoOrderDate { get; set; }
        public string CancelOrderNumber { get; set; }
        public DateTime? CancelOrderDate { get; set; }
        public string CancelReason { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public int PersonId { get; set; }
        public int SanctionTypeId { get; set; }
        public string Description { get; set; }
        public int? InstitutionId { get; set; }
        public short? SchoolYear { get; set; }
        public IEnumerable<DocumentViewModel> Documents { get; set; }
        public byte? SourceType { get; set; }
        public string SchoolYearName { get; set; }
        public string InstitutionName { get; set; }
    }
}
